examples/rsp-next-ts-17/components/Section.tsx

```diff
@@ -3,7 +3,7 @@ import React from "react";
 
 interface SectionProps {
   title: string;
-  children: JSX.Element | JSX.Element[];
+  children: React.JSX.Element | React.JSX.Element[];
```

Comment:
deprecated but not removed. Doesn't really explain why it broke. Any idea?

Reply:
It broke between RAC alpha 4 and alpha 5 https://unpkg.com/browse/react-aria-components@1.0.0-alpha.5/dist/
Here's the list of commits between those two versions https://github.com/adobe/react-spectrum/compare/7186d21...8e6cefd

in alpha 4, we exported using the global namespace which included JSX. in alpha 5, we stopped doing that and started exporting the React namespaced version.

I suspect some parcel or update to types in our own repo which caused us to stop using the global namespace, but I haven't nailed it down yet. Upgrading the types caused it to be marked as deprecated in our own code base, but the problem was already present.

I'm following up with the issue author because I suspect they have multiple copies of react types. I suspect users just need to upgrade their types and dedupe.

Reply:
it "broke" in https://github.com/adobe/react-spectrum/commit/1123b0fe5a6cf12d5172c6b1ef22e84dcf8ddf73
we upgraded our react/types from 17 to 18, and it seems as though our type generation is smart enough to find the non-global namespace of JSX and prefer that. I'm not sure where that is in the parcel transformer?

I haven't actually "fixed" it in this PR, I think people have to update themselves which is what I followed up with the issue author to do.

In addition to explicit references to the global JSX (RAC DialogTrigger), we also have many implicit references to it. Any inferred component's return value for instance. So if we wanted to "fix" this by using the global JSX namespace again, we'd need to handle both of these.